src/TSHelper.ts

```diff
@@ -249,37 +256,43 @@ export class TSHelper {
         return [false, null, null];
     }
 
-    public static isDeclarationWithContext(sigDecl: ts.SignatureDeclaration, checker: ts.TypeChecker): boolean {
+    public static getDeclarationContextType(sigDecl: ts.SignatureDeclaration, checker: ts.TypeChecker): ContextType {
         const thisArg = sigDecl.parameters.find(p => ts.isIdentifier(p.name)
                                                 && p.name.originalKeywordKind === ts.SyntaxKind.ThisKeyword);
         if (thisArg) {
             // Explicit 'this'
-            return !thisArg.type || thisArg.type.kind !== ts.SyntaxKind.VoidKeyword;
+            return thisArg.type && thisArg.type.kind === ts.SyntaxKind.VoidKeyword
+                ? ContextType.Void : ContextType.NonVoid;
         }
         if ((ts.isMethodDeclaration(sigDecl) || ts.isMethodSignature(sigDecl))
             && !(ts.getCombinedModifierFlags(sigDecl) & ts.ModifierFlags.Static)) {
             // Non-static method
-            return true;
+            return ContextType.NonVoid;
         }
         if ((ts.isPropertySignature(sigDecl.parent) || ts.isPropertyDeclaration(sigDecl.parent))
             && !(ts.getCombinedModifierFlags(sigDecl.parent) & ts.ModifierFlags.Static)) {
             // Non-static lambda property
-            return true;
+            return ContextType.NonVoid;
```

Comment:
I thought lambda properties are currently transpiled without context, would that not mean void type here, or do we want to change this?

Reply:
lambda properties get a dummy context (versus stand-alone lambdas, which do not). This ensures they can be called the same as methods when obfuscated behind an interface.